torch/csrc/distributed/c10d/CUDASymmetricMemoryUtils.cpp

```diff
@@ -0,0 +1,223 @@
+#include <sys/socket.h>
+#include <sys/syscall.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+#include <c10/util/error.h>
+
+#if !defined(USE_ROCM) && defined(PYTORCH_C10_DRIVER_API_SUPPORTED)
+#include <c10/cuda/driver_api.h>
+#endif
+
+#include <torch/csrc/distributed/c10d/CUDASymmetricMemoryUtils.hpp>
+#include <torch/csrc/distributed/c10d/Store.hpp>
+#include <torch/csrc/distributed/c10d/cuda/utils.hpp>
+
+namespace c10d::symmetric_memory {
+
+bool device_has_multicast_support(int device_idx) {
+  if (c10::utils::check_env("TORCH_SYMM_MEM_DISABLE_MULTICAST") == true) {
+    return false;
+  }
+  return c10d::cuda::deviceSupportsMulticast(device_idx);
+}
+
+bool allow_overlapping_devices() {
+  return c10::utils::check_env("TORCH_SYMM_MEM_ALLOW_OVERLAPPING_DEVICES") ==
+      true;
+}
+
+// Query environment variable to get the backend used for CUDA Symmetric Memory.
+std::string getSymmMemBackendCUDA() {
+  static auto val = c10::utils::get_env("TORCH_SYMMMEM");
+  if (!val.has_value()) {
+    // In-house implementation: `CUDASymmetricMemory`
+    return "CUDA";
+  } else {
+    // Other backends:
+    //   - "NVSHMEM": `NVSHMEMSymmetricMemory`
+    return val.value();
+  }
+}
+
+IpcChannel::IpcChannel()
```

Comment:
i didn't review the IpcChannel code yet, maybe someone else can take a close look there

Reply:
It is a move. Sorry